Lib/idlelib/grep.py

```diff
@@ -124,9 +174,8 @@ def findfiles(self, dir, base, rec):
         return list
 
     def close(self, event=None):
-        if self.top:
-            self.top.grab_release()
-            self.top.withdraw()
+        "Close the dialog."
+        SearchDialogBase.close(self, event)
```

Comment:
Good catch.  Maybe these were once different.  Now, just delete GrepDialog.close and inherit SearchDialogBase.close.